### PR TITLE
CI: Test Python 3.11

### PR DESCRIPTION
**Pull request by EwoutH**
Modify the Azure pipeline configuration to test on Python 3.11. Remove Python 3.9 and older builds to save CI cycles.

This draft PR is opened to see how the Azure pipelines are performing on Python 3.11, and where the bugs currently are. So for testing only, do not merge.

In support of #8287.

### CODE REVIEW

**Comment by bashtage** [comment]
Can you restore the old Python versions?  These are needed to check the range of supported configs for the next release.  We only update the minimums after a release.

**Comment by EwoutH** [comment]
Thanks for getting back! This PR was never meant to be merged, only for testing.

Would you like a clean PR that only adds a Python 3.11 job?

**Comment by bashtage** [comment]
You can restore them if it is easiest, or a new one.  Up to you.  Thanks.

**Comment by EwoutH** [comment]
Done, I opened a new clean PR: #8431

Could you leave this one open? I would like to keep using it for testing for a while.

File: requirements-dev.txt [hunk 1]
```diff
@@ -23,17 +23,17 @@ flake8
...
-nbsphinx
-nbconvert
+# sphinx
+# sphinx-material
+# jupyter
+# notebook
+# nbsphinx
+# nbconvert
 pyyaml
 pandas-datareader
```
Comment (bashtage) [inline requirements-dev.txt:33]: Perhaps disable this one.

**Comment by EwoutH** [comment]
@bashtage I'm trying some more things on this branch

**Comment by bashtage** [comment]
Hmm.  Doesn't seem to be able to even get Python now.

**Comment by EwoutH** [comment]
Python 3.11.0 stable released last night! I think systems are already updated, and it now want's the stable 3.11 version. I'm trying that in the last commit.

**Comment by EwoutH** [comment]
Python 3.11.0 is now up on the CI (https://github.com/actions/python-versions/pull/193), triggering the CI again.

**Comment by bashtage** [comment]
I marked this as ready.  Is there anything else you need to do?

**Comment by bashtage** [comment]
Going to close since this is superseded by #8431

**Pull request closed without merging**